ndb/src/google/cloud/ndb/key.py

```diff
@@ -1241,3 +1247,11 @@ def _to_legacy_path(dict_path):
         elements.append(element)
 
     return _app_engine_key_pb2.Path(element=elements)
+
+
+class GetByKeyFuture(_future.Future):
+    """Future for looking up entities by Key."""
+
+    def _compute_result(self, result):
+        from google.cloud.ndb import model  # avoid circular import
+        return model._entity_from_protobuf(result.found[0].entity)
```

Comment:
Do you want an assertion that `len(result.found) == 1`? Or that `result.missing` is empty?

Reply:
Yes, there are a number of things to check here. This is just sketchy code.